@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1302,6 +1309,79 @@ export class TextElementsRenderer {
             const layer = textCanvas.getLayer(textElement.renderOrder || DEFAULT_TEXT_CANVAS_LAYER);
 
             const isPathLabel = textElement.path !== undefined && !textElement.isLineMarker;
+            let screenPoints: THREE.Vector2[];
+            let indexOfFirstVisibleScreenPoint = -1;
+
+            if (isPathLabel) {
+                // Get the screen points that define the label's segments and create a path with
+                // them.
+                screenPoints = [];
+                let minX = Number.MAX_SAFE_INTEGER;
+                let maxX = Number.MIN_SAFE_INTEGER;
+                let minY = Number.MAX_SAFE_INTEGER;
+                let maxY = Number.MIN_SAFE_INTEGER;
+                const tileCenterX = textElement.tileCenterX!;
+                const tileCenterY = textElement.tileCenterY!;
+                for (const pt of textElement.path!) {
+                    tempPosition.set(pt.x + tileCenterX, pt.y + tileCenterY, 0);
+                    const screenPoint = this.m_screenProjector.project(tempPosition);
+                    if (screenPoint === undefined) {
+                        continue;
+                    }
+                    screenPoints.push(screenPoint);
+
+                    if (screenPoint.x < minX) {
+                        minX = screenPoint.x;
+                    }
+                    if (screenPoint.x > maxX) {
+                        maxX = screenPoint.x;
+                    }
+                    if (screenPoint.y < minY) {
+                        minY = screenPoint.y;
+                    }
+                    if (screenPoint.y > maxY) {
+                        maxY = screenPoint.y;
+                    }
+
+                    if (indexOfFirstVisibleScreenPoint < 0) {
+                        const firstIndex = screenPoints.findIndex(p2 => {
+                            if (this.m_screenCollisions.screenBounds.contains(p2.x, p2.y)) {
```

Comment:
minor, you could just " return this.m_screenCol..."

Reply:
By popular demand... Done.